### PR TITLE
[PKG-2895] jellyfish 1.0.1

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: **no big changes from 0.9.0** https://github.com/jamesturk/jellyfish/blob/v1.0.1/docs/changelog.md
License: https://github.com/jamesturk/jellyfish/blob/v1.0.1/LICENSE
Requirements: https://github.com/jamesturk/jellyfish/blob/v1.0.1/pyproject.toml

Actions:
1. Clean up the recipe
2. Add rust compiler to build
3. Add maturin to host as a new build backend
4. Update license to MIT
5. Add license_family
6. Add description
7. Update doc_url

Notes:
- `recordlinkage 0.16` https://github.com/AnacondaRecipes/recordlinkage-feedstock/pull/2 requires it

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,31 +1,28 @@
...
 requirements:
   build:
     - {{ compiler('c') }}
+    - {{ compiler('rust') }}
   host:
     - pip
     - python
-    - setuptools
-    - wheel
+    - maturin
```
Comment (cbouss) [inline recipe/meta.yaml:25]:
```suggestion
    - maturin >=0.14,<2
```
imho keeping ranges still makes sense when there is an upper bound

**Review by cbouss — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,31 +1,28 @@
...
   number: 0
-  skip: true  # [py<36]
-  script: {{ PYTHON }} -m pip install . -vv
+  # maturin isn't available on s390x
+  skip: true  # [py<37 or s390x]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
   build:
     - {{ compiler('c') }}
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: Is a C compiler really needed?

**Review by JeanChristopheMorinPerso — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: No, it's not needed. I've checked

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Ah, I've checked on `osx-arm64`, but the C compiler is required on `linux`:
`["  ERROR (jellyfish,lib/python3.8/site-packages/jellyfish/_rustyfish.cpython-38-x86_64-linux-gnu.so): .. but ['libgcc-ng'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)"]`

**Pull request merged**